src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -460,6 +460,21 @@ public void TestHasExited()
             }
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void TestToString_OnExitedProcess()
```

Comment:
Just curios, does this test fail on master branch? With .NET5, this program:

```c#
using System;
using System.Diagnostics;

class Program
{
    static void Main()
    {
        Process p = Process.Start("bash", "-c \"sleep 100000\"");
        Console.WriteLine(p.ProcessName);
        p.Kill();

        Console.WriteLine(p.ToString());
    }
}
```
displays:
```
bash
System.Diagnostics.Process (bash)
```
on Linux and macOS, with return code 0.

Reply:
@am11 sorry for the delay - yes, the above looks _right_, in the test we do: 

```
Assert.True(p.WaitForExit(WaitInMS));
```

In your example, the behaviour is expected, because when you call `.ToString()` the process is actually still there and the runtime is able to get the information - hence no exception and the information in the output.

Reply:
@avodovnik, ah I missed the WaitForExit. With `Process.Start("bash", "--help");` and WaitForExit(10_000), it gives:
```
Unhandled exception. System.InvalidOperationException: Process has exited, so the requested information is not available.
   at System.Diagnostics.Process.EnsureState(State state)
   at System.Diagnostics.Process.get_ProcessName()
```
however, with subprocess `Process.Start("bash", "-c \"sleep 1\"")` and WaitForExit(10_000), it strangely returns subprocess name instead:
```
sleep
System.Diagnostics.Process (sleep)
```
does this PR change/improve this behavior?

Reply:
Yes, the PR follows our own guidelines on not throwing exceptions in `ToString` and instead falls back to the default value without displaying the process name. As for the `sleep` case, the returned name is a result of whatever the underlying process manager returns for that PID, and I _believe_ `sleep` is correct - it is, in fact, the current process running under that id.